apps/ledger-live-desktop/src/renderer/modals/SignTransaction/steps/StepConnectDevice.tsx

```diff
@@ -1,22 +1,43 @@
-import React from "react";
+import React, { useMemo } from "react";
 import TrackPage from "~/renderer/analytics/TrackPage";
 import { getMainAccount } from "@ledgerhq/live-common/account/index";
 import GenericStepConnectDevice from "./GenericStepConnectDevice";
 import { StepProps } from "../types";
+
+function dependenciesToAppRequests(dependencies?: string[]) {
```

Comment:
[COULD] Add a return type here to tell that you've known what has been done as the map changes the structure and the parameter name has been used as property name, this typing can make it more clear.
In addition, returning an empty array is compatible to the return type annotation and can help to avoid the `||` operator afterwards.
```suggestion
function dependenciesToAppRequests(dependencies?: string[]): { appName: string } {
```

Reply:
Done

Reply:
But I think you should keep the same behaviour at `apps/ledger-live-mobile/src/screens/SignTransaction/02-ConnectDevice.tsx` line 21.

Reply:
Mobile just doesn't have the case where we don't have an account, that's why they differ a bit